### PR TITLE
feat: CICD to firebase

### PR DESCRIPTION
**Pull request by mmcknett**
This change:

1. Adds deployment to firebase when merging to `main`
2. Adds deployment to a preview channel when creating a PR
3. Cleans up dev dependencies a little.
4. Fix lint errors that apparently only show up in CI (because the `CI` environment variable is set to `true`)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,22 +5,10 @@
...
-    "@types/jest": "^26.0.15",
-    "@types/node": "^16.0.0",
-    "@types/react": "^16.9.53",
-    "@types/react-alert": "^5.2.0",
-    "@types/react-dom": "^16.9.8",
-    "@types/styled-components": "^5.1.7",
     "firebase": "^8.2.4",
     "firestore-export-import": "^0.9.0",
     "google-map-react": "^2.1.9",
-    "install": "^0.13.0",
```
Comment (anyatokar) [inline package.json:21]: not sure what "install" was used for? assuming you deleted it on purpose!

**Review by anyatokar — APPROVED** [review]
Ah yay! Thank you for adding the scripts/cleaning this up! 
I merged my own PR #44 which also affected the package.json and therefore created a conflict and due to the fact that I did a terrible thing I took it upon myself to resolve the conflicts. Many apologies.

File: package.json (hunk 1 above)
Comment (mmcknett) [reply package.json:21]: I couldn't find it being used anywhere, so I removed it. I'm not sure what it was supposed to be for, either!

File: .github/workflows/mfte-simple-deploy.yml
[the remark this answers is not in the document]
Comment (mmcknett) [reply .github/workflows/mfte-simple-deploy.yml:1]: I guess it isn't strictly necessary? Do you want me to remove it?

**Comment by mmcknett** [comment]
I rebased on your latest `main`!

File: .github/workflows/mfte-simple-deploy.yml
[the remark this answers is not in the document]
Comment (anyatokar) [reply .github/workflows/mfte-simple-deploy.yml:1]: at time of writing it was not there :D no need to do anything

**Pull request merged**